lib/board.rb

```diff
@@ -1,33 +1,63 @@
 class Board 
-	attr_accessor :matrix
-	def initialize
-		@matrix = [nil, nil, nil, nil, nil, nil, nil, nil, nil]
+	attr_accessor :matrix, :win_combinations, :dim
+	def initialize(dim)
+		@matrix = Array.new(dim**2)
+		@win_combinations = init_combin(dim)
+		@dim = dim
+	end
+  
+  	def init_combin(dim)
+  		arr = []
+        arr  += Array.new(dim ){|l| Array.new(dim){|i| dim * l + i }}  
+        arr  += Array.new(dim){|l| Array.new(dim){|i|  l + (i*dim)}}
+        arr  += Array.new(1){|l| Array.new(dim){|i|  i+ (i*dim)}}
+        arr  += Array.new(1){|l| Array.new(dim){|i|  i*(dim-1) + (dim-1)}}
+        return arr                      
+    end
+
+    def valid_input?(arr)
+		if arr.length == 2 && 
+		(ALPHABET[0...@dim].any? {|val| val == arr[1].upcase}) &&
+		((0...@dim).to_a.any? {|element| element == arr[0].to_i})
+			return true
+		end
+		false
+	end
```

Comment:
You never use this method inside the `board` object. You could validate every move before it is written or you could put the validator outside and assume you are getting a valid input. Either way is fine. Just pick one strategy, not half validation in one object and the other half on another. (If you assume you are getting a valid input that doesn't mean you shouldn't protect against possible errors like index out of reach).